addons/sms_twilio/models/sms_sms.py

```diff
@@ -0,0 +1,38 @@
+from odoo import fields, models
+
+
+class SmsSms(models.Model):
+    _inherit = 'sms.sms'
+
+    sms_twilio_sid = fields.Char(related="sms_tracker_id.sms_twilio_sid", depends=['sms_tracker_id'])
+
+    failure_type = fields.Selection(
+        selection_add=[
+            ('sms_twilio_authentication', 'Authentication error')
+        ],
+        ondelete={'sms_twilio_authentication': 'cascade'},
+    )
+
+    def send(self, unlink_failed=False, unlink_sent=True, auto_commit=False, raise_exception=False):
+        if self.env.company.sms_provider != 'twilio':
+            return super().send(unlink_failed=unlink_failed, unlink_sent=unlink_sent, auto_commit=auto_commit, raise_exception=raise_exception)
+        # Call super method with auto_commit = True
+        return super().send(unlink_failed=unlink_failed, unlink_sent=unlink_sent, auto_commit=True, raise_exception=raise_exception)
+
+    def _get_batch_size(self):
+        if self.env.company.sms_provider != 'twilio':
+            return super()._get_batch_size()
+        return int(self.env['ir.config_parameter'].sudo().get_param('sms_twilio.session.batch.size', 10))
+
+    def _handle_call_result_hook(self, results):
+        # Store the sid of Twilio on the SMS tracking record (as SMS will be deleted)
+        if self.env.company.sms_provider != 'twilio':
+            return
+        grouped_self = self.grouped("uuid")
+        for result in results:
+            sms = grouped_self.get(result.get('uuid'))
+            if sms and sms.sms_tracker_id and result.get('sms_twilio_sid'):
+                sms.to_delete = False  # Let the webhook update this
```

Comment:
`to_delete` is already handled with `unlink_sent` in `_send`, no? Why should twilio handle these differently?

The only thing remaining to do is the extra line you've already added in the controller :)